### PR TITLE
IdeaPage Tab Overhaul/Overflow Control

### PR DESCRIPTION
**Pull request by alanlgirard**
Overhauled the idea page tabs to look a little better and to wrap properly.  The updates and comments now show as if they are dialogues. The backing tabs and team tabs were re-factored to take on the same kind of look.  

Update/Comments Page
image

Backers Tab
image

Team Tab
image

Overflow Areas
* All ideasView tabs
* Add/Edit idea description
* Add/Edit Back details
* Edit Team chips
* Leave Team button placement on small screens

As a bonus I fixed some of the form validation that was not working correctly before (back description) and made it so the login doesn't "bump" down anymore because the ng-messages were only showing on the touched state.

Comment/Approve away @Mctalian @monotkate @YashdalfTheGray @bcbrennecke

### CODE REVIEW

**Comment by alanlgirard** [comment]
Will close issue #166

**Comment by Mctalian** [comment]
Weren't we able to delete comments before? Or am I crazy?

**Comment by alanlgirard** [comment]
You can still delete comments.  I got rid of the hover-over effect and now the delete comment is the x in the upper right of each comment/update

**Comment by Mctalian** [comment]
Also, I think a little extra padding between the text box at the top of updates/comments and the first update/comment would help. Not too much, just a bit.

Overall, I like it. There are some things that feel a bit off but it's a major improvement over the old stuff. I'll try to put those things into words soon. Nice work!

**Comment by monotkate** [comment]
- We need a way to highlight the controls at the top now.  Potentially their own card to box them in?
- I'm not a huge fan of the team tab restructure.  Potentially split it into the users like the other tabs instead of all of them in the same team area?  Also, I think we should split the email into single users then (although really I think this should only be an option if you're already on the team).  
- I don't like the X next to the name either, but if you restructure it a bit more like the comments, I think it could match them and look a lot better.
- More padding above for updates/comments just like @Mctalian said.  It's cutting it a but close.  Or, just add padding to the bottom of the entry. That would work too.

+ Overall a huge fan of the restructuring.  I like how the controls are embedded in each portion :100: It feels way more modernized this way.

**Comment by YashdalfTheGray** [comment]
This...this...my dear lord this looks good! I love the embedded controls, totally fits in with Material Design as well!

File: src/assets/app.css [hunk 1]
```diff
@@ -300,3 +300,82 @@ img.grayscale:hover {
...
+}
+
+.triangle-isosceles.left:after {
+  top:16px; /* controls vertical position */
+  left:-10px; /* value = - border-left-width - border-right-width */
+  bottom:auto;
+  border-width:10px 10px 10px 0;
+  border-color:transparent #e0e0e0;
+}
+
```
Comment (YashdalfTheGray) [inline src/assets/app.css]: Check your spacing on the `.triangle-isosceles`, `.triangle-isosceles.left`, `.triangle-isosceles:after` and `.triangle-isosceles.left:after` classes. I think you've got 2 spaces.

**Comment by YashdalfTheGray** [comment]
I think all the team members being in one thing makes sense since they're all related (since they are on a team together). Maybe see if you can get the 'x's to be aligned to the right? I think that would fit better with Material Design guidelines anyway. 

More padding could be nice, would probably look better, I'm not too opposed to the way it is right now though.

**Comment by davethenipper** [comment]
-trashcan for deleting
-team members being separate

**Pull request merged**